src/SchemaParser.f90

```diff
@@ -0,0 +1,1551 @@
+      TYPE(StringType),INTENT(IN) :: pListPathCrd
+    ENDSUBROUTINE parse_SchEnt_absintfc
+  ENDINTERFACE
+
+  !> Exception Handler for use in SchemaParser
+  TYPE(ExceptionHandlerType),SAVE :: eSchemaParser
+!
+!===============================================================================
+  CONTAINS
+!
+!-------------------------------------------------------------------------------
+!> @brief Constructor for the schema parser
+!> @param this the variable to initialize
+!>
+!> The constructor for the schema parser
+!>
+    SUBROUTINE init_SchPar(this)
+      CHARACTER(LEN=*),PARAMETER :: myName='init_SchPar'
+      CLASS(SchemaParserType),INTENT(INOUT) :: this
+
+      ALLOCATE(this%block(0))
+      this%isInit=.TRUE.
+    ENDSUBROUTINE init_SchPar
+!
+!-------------------------------------------------------------------------------
+!> @brief Routine clears the data in Schema Parser type variable
+!> @param this the type variable to clear
+!>
+    SUBROUTINE clear_SchPar(this)
+      CLASS(SchemaParserType),INTENT(INOUT) :: this
+
+      INTEGER(SIK) iblock,nBlocks
+
+      REQUIRE(this%isInit)
+      nBlocks=SIZE(this%block)
+      DO iblock=1,nBlocks
+        CALL this%block(iblock)%clear()
+      ENDDO
+      DEALLOCATE(this%block)
+      this%isInit=.FALSE.
+    ENDSUBROUTINE clear_SchPar
+!
+!-------------------------------------------------------------------------------
+!> @brief initializes this schema element
+!> @param this        the element to be initialized
+!> @param name        the name of the element as it will appear in the input
+!> @param pListPath   the parameter list path under which the element data will
+!>                    put in the outgoing parameter list
+!> @param type        occurence type, either single or multi occurrence
+!> @param required    whether or not the element is required`
+!>
+    SUBROUTINE init_SchElm(this,name,pListPath,type,required)
+      CLASS(SchemaElementType),INTENT(INOUT) :: this
+      TYPE(StringType),INTENT(IN) :: name
+      TYPE(StringType),INTENT(IN) :: pListPath
+      INTEGER(SIK),INTENT(IN) :: type
+      LOGICAL(SBK),INTENT(IN) :: required
+
+      ! Create the element
+      this%name = TRIM(name)
+      this%pListPath = TRIM(pListPath)
+      this%type = type
+      this%isRequired = required
+    ENDSUBROUTINE init_SchElm
+!
+!-------------------------------------------------------------------------------
+!> @brief States whether or not the element has the provided name
+!> @param this  the element whose name is to be checked
+!> @param name  the name 
+!>
+    FUNCTION hasName_SchElm(this,name) RESULT(hasName)
+      CLASS(SchemaElementType),INTENT(IN) :: this
+      TYPE(StringType),INTENT(IN) :: name
+      LOGICAL(SBK) :: hasName
+
+      hasName=(TRIM(name)==TRIM(this%name))
+    ENDFUNCTION hasName_SchElm
+!
+!-------------------------------------------------------------------------------
+!> @brief Determines the number of times an element occurs in a given file
```

Comment:
Yeah, Ben was recommending making the addCard and addBlock subroutines into functions which return block / card handlers, which would bypass the need for string comparisons.

Ultimately, FORTRAN needs to do better string comparisons (#weaksauce)

Reply:
Talked offline and there are ideas for `include`ing a file, which addresses point 2.  Point 1 we'll just have to wait and find out.  I doubt it will be an issue, and we can optimize things if there are.